### ISSUE TITLE
uffi-compat: convert-from-foreign-string can never return an empty string

### ISSUE THREAD

**Issue by dieggsy** [cffi/cffi#389]
I find this a bit weird. There are cases when what you actually want returned is an empty string.

This shows up when trying to use CLSQL with CLSQL-CFFI instead of the UFFI default. UFFI handles this correctly, but CFFI's uffi-compat does not.

**Comment by luismbo** [comment]
Yes, indeed. Perhaps the UFFI API behaved like that at the time? I don't recall. Would you like to have a go at fixing it? Might be as simple as removing the ìf` form?
